test/jdk/java/awt/TrayIcon/TrayIconScalingTest.java

```diff
@@ -0,0 +1,126 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8255439
+ * @key headful
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary To test tray icon scaling with on-the-fly DPI/Scale changes on Windows
+ * @run main/manual TrayIconScalingTest
+ * @requires (os.family == "windows")
+ */
+
+import java.awt.font.TextLayout;
+import java.awt.image.BaseMultiResolutionImage;
+import java.awt.image.BufferedImage;
+import java.awt.*;
+import java.awt.image.MultiResolutionImage;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+
+public class TrayIconScalingTest {
+
+    private static final String INSTRUCTIONS =
+            "This test case checks the scaling of tray icons for on-the-fly" +
+                    " DPI/ Scale changes on Windows.\n\n" +
+                    "STEPS: \n\n" +
+                    "1. When you run this test check the system tray/" +
+                    " notification area on windows, a white multi-resolution" +
+                    " image (MRI) icon should be visible.\n\n"+
+                    "2. Navigate to Settings > System > Display and change the" +
+                    " display scale by selecting any value from" +
+                    " Scale & Layout dropdown.\n\n"+
+                    "3. On scale changes observe the white tray icon," +
+                    " if there is NO distortion then press PASS.\n\n";
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
+        PassFailJFrame passFailJFrame = new PassFailJFrame("TrayIcon " +
+                "Test Instructions", INSTRUCTIONS, 8, 18, 85);
+        createAndShowGUI();
+        passFailJFrame.awaitAndCheck();
+
+    }
+
+    private static void createAndShowGUI() {
+
+        // Check if SystemTray supported on the machine
+        if (!SystemTray.isSupported()) {
+            System.out.println("SystemTray is not supported");
+            return;
+        }
```

Comment:
> This should be the first statement in main. If icons aren't supported, there's no need to create the instruction frame. The test will still wait for user's response but no tray icon is added.

It won't wait if you haven't created it .. it will just exit without showing anything and be counted as pass, that's probably OK.

Reply:
Added the **!SystemTray.isSupported()** check as first line in main(). In case the system tray is not supported, the test is counted as pass and does not proceed to display instruction frame.